src/components/card.js

```diff
@@ -1,31 +1,57 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._data = {
-      name: data.name,
-      link: data.link,
-    };
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
+    this._name = data.name;
+    this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
+
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
     this._cardSelector = cardSelector;
   }
 
   _setEventListeners() {
-    this._likeIcon.addEventListener("click", this._handleLikeIcon);
-
-    this._deleteBtn.addEventListener("click", this._handleDeleteCard);
-
+    this._likeIcon.addEventListener("click", () => {
+      this._handleLikeClick(this);
+    });
+    this._deleteBtn.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
     this._cardImageEl.addEventListener("click", () => {
-      this._handleImageClick(this._data);
+      this._handleImageClick({
+        name: this._name,
+        link: this._link,
+      });
     });
   }
+  setIsLiked(isLiked) {
+    this.isLiked = isLiked;
+    this.renderLikes();
+  }
 
-  _handleLikeIcon = () => {
+  renderLikes() {
+    if (this.isLiked) {
+      this._likeIcon.classList.add("card__react-button_active");
+    } else {
+      this._likeIcon.classList.remove("card__react-button_active");
+    }
+  }
+
+  toggleLike() {
```

Comment:
Needs correction: please remove this method and it's usages as you have `renderLikes` method for changing the likes button state according to the isLiked flag